static/js/containers/CourseSearchPage.js

```diff
@@ -0,0 +1,322 @@
+// @flow
+/* global SETTINGS: false */
+import qs from "query-string"
+import R from "ramda"
+import React from "react"
+import InfiniteScroll from "react-infinite-scroller"
+import { connect } from "react-redux"
+import { MetaTags } from "react-meta-tags"
+import _ from "lodash"
+import type { Location, Match } from "react-router"
+import type { Dispatch } from "redux"
+import Checkbox from "rmwc/Checkbox/index"
+
+import CanonicalLink from "../components/CanonicalLink"
+import Card from "../components/Card"
+import { Cell, Grid } from "../components/Grid"
+import {
+  CourseSearchLoading,
+  Loading,
+  PostLoading,
+  withLoading
+} from "../components/Loading"
+import SearchTextbox from "../components/SearchTextbox"
+import SearchResult from "../components/SearchResult"
+
+import { actions } from "../actions"
+import { clearSearch } from "../actions/search"
+import { COURSE_AVAILABILITIES } from "../lib/constants"
+import { SEARCH_FILTER_COURSE } from "../lib/picker"
+import { preventDefaultAndInvoke, toArray } from "../lib/util"
+import { validateSearchQuery } from "../lib/validation"
+
+import type {
+  SearchInputs,
+  SearchParams,
+  Result,
+  CourseFacetResult
+} from "../flow/searchTypes"
+
+type Props = {
+  dispatch: Dispatch<any>,
+  location: Location,
+  history: Object,
+  initialLoad: boolean,
+  isModerator: boolean,
+  match: Match,
+  runSearch: (params: SearchParams) => Promise<*>,
+  results: Array<Result>,
+  searchLoaded: boolean,
+  loaded: boolean,
+  searchProcessing: boolean,
+  total: number,
+  clearSearch: () => void,
+  facetChoices: CourseFacetResult,
+  facetChoiceProcessing: boolean
+}
+type State = {
+  text: string,
+  topics: Array<string>,
+  platforms: Array<string>,
+  availabilities: Array<string>,
+  from: number,
+  error: ?string
+}
+
+export class CourseSearchPage extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props)
+    this.state = {
+      text:           qs.parse(props.location.search).q,
+      topics:         qs.parse(props.location.search).t,
+      platforms:      qs.parse(props.location.search).p,
+      availabilities: qs.parse(props.location.search).a,
+      from:           0,
+      error:          null
+    }
+  }
+
+  componentDidMount() {
+    const { text } = this.state
+    const { clearSearch, loaded, facetChoiceProcessing } = this.props
+    clearSearch()
+    if (!loaded && !facetChoiceProcessing) {
+      this.loadFacetChoices()
+    }
+    if (text) {
+      this.runSearch()
+    }
+  }
+
+  loadFacetChoices = async () => {
+    const { dispatch } = this.props
```

Comment:
It looks like this function is uncovered. Can you write a test which uses its functionality?

Reply:
Done